For Qt5 the header files of widgets have to be explicitly included.

--- a/dialogbox.hpp	2016-03-02 20:31:01.000000000 +0100
+++ b/dialogbox.hpp	2016-12-27 21:54:05.335465144 +0100
@@ -22,6 +22,23 @@
 #define DIALOGBOX_H
 
 #include <QtGui>
+#include <QApplication>
+#include <QDir>
+#include <QFile>
+#include <QDialog>
+#include <QTabWidget>
+#include <QListWidget>
+#include <QBoxLayout>
+#include <QLayoutItem>
+#include <QLabel>
+#include <QCheckBox>
+#include <QGroupBox>
+#include <QRadioButton>
+#include <QPushButton>
+#include <QProgressBar>
+#include <QComboBox>
+#include <QLineEdit>
+#include <QTextEdit>
 
 #define BUFFER_SIZE 1024
 
